### ISSUE TITLE
New Resource Manager UI Enhancements

### ISSUE THREAD

**Issue by jeremy-farrance** [dnnsoftware/Dnn.Platform#5363]
## Description of problem

**1. UI improvement for Refresh/Sync**

Working with files in Global Assets I was adding files on the server directly to /Portals/_default/ and I used Sync, both Refresh and "Sync This Folder & Subfolders" a few times and found that they gave no indication that they were working. I did test and verify by adding folders/files on the server in to Global Assets root, then doing Sync, the folders and files do appear as expected. But the UI gives no indication that work is being done. 

**2. Expected Behavior for Folders**

I noticed one additional thing that felt like a bug, but is more of a needed feature enhancement... in the right folder/files view it seemed very unexpected that double-clicking a folder did not take you *in* to (open) that folder. Along the same lines, right-clicking a folder, the context menu does not contain an option to Open. 

## Description of solution
1. For the first problem (Sync), there are many possible solutions. The best solution probably depends on whether the process is actually handled synchronously or async. 

If async, then UI still needs to at least do something 200-400ms to indicate something happened, the expected work was done (even if nothing changed in the view). My simple suggestion would be to dim the right-hand panel of the split view for a moment, a half second at most.

If the refresh/sync is synchronously, then dim panel the entire time its running, maybe add an icon-spinner or simple progress bar, and when done, undim and reloads the right folder/file view.
 
2. Add both expected UI abilities to folders in the right view, double-click or right-click and open should take you into that folder. 

## DNN v9.11.00, affected browsers, all

**Comment by david-poindexter** [comment]
@jeremy-farrance first of all, thanks for posting this!  

One request for future reference...please break out each bug or feature request into its own issue.  This will make these clean and actionable.

1. This one is a tough one and I believe a double-edged sword.  The focus for the 9.11.0 release was feature parity and this works now the same way it did (be that right or wrong).  That said, these actions are super fast, so doing a "processing" type of user feedback would have to be faked (significant time delay) just to have it up long enough to see.  I can see the value in this for sure and good UX should win in the end.  As a developer, it just feels wrong.  LOL

2. The double-click to open a folder has already been resolved in #5352 thanks to @valadas.  🎉 The right-click > Open is also a good idea.

**Comment by valadas** [comment]
I have renamed the title of this issue, since it reported 2 things but one is done already...

**Comment by valadas** [comment]
This is in my list for when I have some free time...

**Comment by stetard** [comment]
Regarding the UI improvement, I would suggest to consider the download file action too. Some users told me that they would appreciate if the UI could indicate that the file download is running. It could take more than 10 secondes before the browser displays something when the requested file is big (more over 100 Mb).
Do I need to create another issue or is it possible to group that request with the current one?

**Comment by jeremy-farrance** [comment]
Yes, I think that should be its own issue. Are you able to provide more detail on how you expect it to work? I think a screenshot(s) of what it looks like now and be more specific about where+when the problem is at its worst. Maybe record a show screen video demo?
